### ISSUE TITLE
Whitelist jekyll-data plugin

### ISSUE THREAD

**Issue by yatil** [github/pages-gem#610]
In lieu of a proper translation solution (#401), can we at least make data in plugins portable by whitelisting https://github.com/ashmaroli/jekyll-data. Keeping data that belongs to the theme (like common navigation or translation phrases) in sync is really complicated at the moment. This would be a relatively simple solution.

**Comment by big-andy-coates** [comment]
Yep, this would be really useful, e.g. to maintain a consistent navigation menu when a site is built by composing the gh pages of multiple gh repos.

Currently, I have to keep data files synchronised across 20 odd repos.  A right PITA! ;)

**Comment by big-andy-coates** [comment]
Related to https://github.com/benbalter/jekyll-remote-theme/issues/68

**Comment by parkr** [comment]
We upstreamed theme data folder propagation to Jekyll: https://github.com/jekyll/jekyll/pull/8815

I believe this repo is in maintenance mode. GitHub Actions now has first-class support for Pages, so I'd recommend setting that up for affected repos if you can.

**Comment by parkr** [comment]
Use Jekyll v4.3.0 or later. Full release notes (including the above linked PR): https://github.com/jekyll/jekyll/releases/tag/v4.3.0

**Comment by big-andy-coates** [comment]
Thanks!

**Comment by big-andy-coates** [comment]
A bit of Googling on this seems to suggest that this involves switching to a workflow to build the docs, and it may well be the case that the thème I'm using isn't compatible with Jekyll 4.x.

I'm therefore wondering if this whitelisting would enable the required functionally without the need to switch to workflow or finding a Jekyll 4.x theme.

Thoughts @parkr ?

**Comment by parkr** [comment]
@big-andy-coates The Jekyll v4 upgrade isn't that big actually – the reason I never got around to it while I worked at GitHub was because there was a breaking change in `post_url` which would have silently broken millions of sites. But if you're doing the upgrade yourself, you can check on links throughout the site to ensure they're correct. Jekyll even has upgrading docs: https://jekyllrb.com/docs/upgrading/3-to-4/

I don't work for GH anymore but from what I gather on their blog, Actions is The Future ™️  so they're likely not accommodating the security review and maintenance of additional plugins at this time. I can't say for sure though, but if I were you, I'd migrate to an (Actions) workflow and have full customizability. :)
